lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -6975,13 +6975,22 @@ class PullbackEmitter final : public SILInstructionVisitor<PullbackEmitter> {
           addToAdjointBuffer(bb, origArg, tmpBuf, loc);
           builder.emitDestroyAddrAndFold(loc, tmpBuf);
           builder.createDeallocStack(loc, tmpBuf);
-        }
-        else {
+        } else {
           recordTemporary(tan);
           addAdjointValue(bb, origArg, makeConcreteAdjointValue(tan), loc);
         }
       }
     }
+    // Destroy unused pullback direct results. Needed for pullback results from
+    // VJPs extracted from `@differentiable` function callees, where the
+    // `@differentiable` function's differentiation parameter indices are a
+    // superset of the active `apply` parameter indices.
+    while (allResultsIt != allResults.end()) {
+      auto unusedPullbackDirectResult = *allResultsIt++;
+      if (unusedPullbackDirectResult->getType().isAddress())
+        continue;
+      builder.emitDestroyOperation(loc, unusedPullbackDirectResult);
```

Comment:
```suggestion
      builder.emitDestroyValueOperation(loc, unusedPullbackDirectResult);
```
`SILBuilder::emitDestroyOperation` has a check and skips addresses, which you've already done above the call. So ideally you should call `SILBuilder::emitDestroyValueOperation` instead.

Reply:
Okay! Done in https://github.com/apple/swift/pull/28207/commits/9783b0adf07f78ca98351909141099bca4be8c52.